.github/workflows/package.yml

```diff
@@ -2,99 +2,79 @@ name: Package
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.cargo/git
-          ~/.cargo/registry
-          target
-        key: ${{ runner.os }}-package-cargo-${{ hashFiles('**/Cargo.lock') }}
-    - name: Cache cargo-deb and cross
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.cargo/bin/cargo-deb
-          ~/.cargo/bin/cross
-          ~/.cargo/.crates.toml
-          ~/.cargo/.crates2.json
-        key: ${{ runner.os }}-cargo-bin-${{ env.cargo-deb-version }}
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: sudo apt-get install libdbus-1-dev ${{ matrix.binutils }}
+      - name: Toolchain
+        if: matrix.target != ''
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+          override: true
 
-    - name: Install cargo-deb
-      run: cargo install cargo-deb --version ${{ env.cargo-deb-version }}
+      - name: Cache .cargo and target
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/git
+            ~/.cargo/registry
+            target
+          key: ${{ runner.os }}-package-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Cache cargo-deb and cross
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/cargo-deb
+            ~/.cargo/bin/cross
+            ~/.cargo/.crates.toml
+            ~/.cargo/.crates2.json
+          key: ${{ runner.os }}-cargo-bin-${{ env.cargo-deb-version }}
 
-    - name: Cross build
-      uses: actions-rs/cargo@v1
-      with:
-        use-cross: true
-        command: build
-        args: --release --target ${{ env.target }}
-    - name: Package
-      run: cd mijia-homie && cargo deb --target ${{ env.target }} --no-build
-    - name: Upload package
-      uses: actions/upload-artifact@v2
-      with:
-        name: debian-packages
-        path: target/${{ env.target }}/debian/
-
-  build:
-    runs-on: ubuntu-latest
-
-    steps:
-    - uses: actions/checkout@v2
-    - name: Install dependencies
-      run: sudo apt-get install libdbus-1-dev
+      - name: Install cargo-deb
+        run: cargo install cargo-deb --version ${{ env.cargo-deb-version }}
 
-    - name: Cache .cargo and target
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.cargo/git
-          ~/.cargo/registry
-          target
-        key: ${{ runner.os }}-package-cargo-${{ hashFiles('**/Cargo.lock') }}
-    - name: Cache cargo-deb and cross
-      id: cache-cargo-deb
-      uses: actions/cache@v2
-      with:
-        path: |
-          ~/.cargo/bin/cargo-deb
-          ~/.cargo/bin/cross
-          ~/.cargo/.crates.toml
-          ~/.cargo/.crates2.json
-        key: ${{ runner.os }}-cargo-deb-${{ env.cargo-deb-version }}
+      # If a target is set, build with cross.
+      - name: Cross build
+        if: matrix.target != ''
+        uses: actions-rs/cargo@v1
```

Comment:
If I disable my docker daemon and run `cross build` with no arguments or `cross build --target x86_64-apple-darwin` then it Just Works (the former puts stuff in ./target/debug and the latter in ./target/x86_64-apple-darwin/debug)

I think that our build scripts would be greatly simplified if we use cross for all targets in the matrix. It may take longer to install the extra toolchain, but I think that's fine.

Reply:
This doesn't seem to work for me; it runs in the docker container `rustembedded/cross:x86_64-unknown-linux-gnu-0.2.1` and fails.